pkg/dns/helpers.go

```diff
@@ -33,7 +33,9 @@ func ToRecordRequestsAndHash(domain string, ingress *v1.Ingress) ([]RecordReques
 
 	var hosts []string
 	for _, rule := range ingress.Spec.Rules {
-		if strings.HasSuffix(rule.Host, domain) {
+		if rule.Host == strings.TrimPrefix(domain, ".") {
```

Comment:
What if we simplified this so that the only thing this could create was a wild card record...so basically hosts would only ever be of size 1 and the check would be like
```
f strings.HasSuffix(rule.Host, domain) {
			hosts = append(hosts, "*")
			break
		}
```

Reply:
Makes sense to me.